GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -192,29 +193,40 @@ public void Add(GitRevision revision, RevisionNodeFlags types, bool insertAsFirs
             bool updateParents = true;
             if (!_nodeByObjectId.TryGetValue(revision.ObjectId, out RevisionGraphRevision revisionGraphRevision))
             {
-                int score = insertAsFirst
+                // This revision is added from the log, but not seen before. This is probably a root node (new branch)
+                // OR the revisions are not in topo order. If this the case, we deal with it later.
+                int score = _maxScore;
 
-                    // Insert first artificial (WorkTree) before first existing
-                    ? -1
+                if (insertScore is not null && _nodeByObjectId is not null)
+                {
+                    // This revision is to be inserted before a certain node
+                    foreach (var kvp in _nodeByObjectId)
+                    {
+                        if (kvp.Value.Score < insertScore)
+                        {
+                            // Insert reserved scores
+                            kvp.Value.OffsetScore(-insertRange);
+                        }
+                    }
```

Comment:
You can use a deconstructor, which makes it easier to read. Though you'd have to give it a better variable name 😉:
```suggestion
                    foreach (var (_, revisionGraphRevision1) in _nodeByObjectId)
                    {
                        if (revisionGraphRevision1.Score < insertScore)
                        {
                            // Insert reserved scores
                            revisionGraphRevision1.OffsetScore(-insertRange);
                        }
                    }
```

Reply:
graphRev